cmd/cli_test.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/NethermindEth/1click/configs"
+	"github.com/NethermindEth/1click/internal/pkg/commands"
+	"github.com/NethermindEth/1click/test"
+	log "github.com/sirupsen/logrus"
+)
+
+type cliCmdTestCase struct {
+	configPath     string
+	generationPath string
+	runner         commands.CommandRunner
+	fdOut          *bytes.Buffer
+	args           []string
+	isErr          bool
+}
+
+func resetCliCmd() {
+	cfgFile = ""
+	executionName = ""
+	consensusName = ""
+	validatorName = ""
+	generationPath = configs.DefaultDockerComposeScriptsPath
+	randomize = false
+	install = false
+	run = false
+	y = false
+	services = &[]string{}
+}
+
+func buildCliTestCase(t *testing.T, caseName string, args []string, isErr bool) *cliCmdTestCase {
+	tc := cliCmdTestCase{}
+	configPath := t.TempDir()
+
+	err := test.PrepareTestCaseDir(filepath.Join("testdata", "cli_tests", caseName, "config"), configPath)
+	if err != nil {
+		t.Fatalf("Can't build test case: %v", err)
+	}
+	dcPath := filepath.Join(configPath, "docker-compose-scripts")
+	err = os.Mkdir(dcPath, os.ModePerm)
+	if err != nil {
+		t.Fatalf("Can't build test case: %v", err)
+	}
+
+	// TODO: allow runner edition
```

Comment:
I don't follow this TODO, please explain

Reply:
The runner here refers to the command runner mocks. In this case this is prefixed for every test case built with `buildCliTestCase`, so the TODO is for allowing it's edition.